### ISSUE TITLE
Add search benchmarks

### ISSUE THREAD

**Issue by LilithHafner** [JuliaDocs/Documenter.jl#2417]
It would be nice to have benchmarks to track search performance (i.e. ensuring that good results are ranked highly)

For example

Base.push! should rank above LibGit2.push! for the search query "push!" on docs.julialang.org. (it currently does not)

Feel free to contribute more candidate benchmarks to this thread. Ideally we could get them to run automatically in CI when folks propose changes to the searching/indexing functionality.

**Comment by mortenpi** [comment]
This seems like a good idea, if there's anyone who is willing to develop the infra and maintain it.

**Comment by LilithHafner** [comment]
Is the existing test-system documented anywhere? If so, and if there are already end to end tests that navigate built html documentation with a headless browser, then it should be pretty easy to add these tests.

**Comment by mortenpi** [comment]
> Is the existing test-system documented anywhere?

No, not really, you'd have to follow the code. The test builds live in `test/examples` though, with the main entry points being `test/examples/make.jl` for building and `test/examples/test.jl` for doing some basic checks on the generated outputs. Any sort of docs/comments/refactorings are always welcome in the test code though.
